pilot/pkg/security/authz/model/generator.go

```diff
@@ -116,7 +118,16 @@ func (srcNamespaceGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission
 }
 
 func (srcNamespaceGenerator) principal(_, value string, forTCP bool) (*rbacpb.Principal, error) {
-	v := strings.Replace(value, "*", ".*", -1)
+	var v string
+	if features.EnableAuthzRegexMatching && strings.HasPrefix(value, "regex:") {
+		v = strings.TrimPrefix(value, "regex:")
+	} else {
+		v = strings.Replace(value, "*", ".*", -1)
+	}
+
+	if err := validation.ValidateRegex(v); err != nil {
```

Comment:
We should not be doing validation on the config generation path. This is extremely expensive. Should validate it in the validation only.

Reply:
If we only validate the regex in validation, where should I call the validation.validateRegex(v) ?